src/compiler/utilities.ts

```diff
@@ -517,12 +517,46 @@ namespace ts {
 
         if (isJSDocTypeExpressionOrChild(node)) {
             // strip space + asterisk at line start
-            text = text.replace(/(^|\r?\n|\r)\s*\*\s*/g, "$1");
+            text = text.split(/\r\n|\n|\r/).map(l => trimStringStart(l.replace(/^\s*\*/, ""))).join("\n");

```

Comment:
I have to admit that `l` is not my favorite variable name - it took me a second to figure out why you were calling `replace` on a numeric literal.